src/vivarium/interpolation.py

```diff
@@ -1,120 +1,324 @@
 import warnings
 
 import pandas as pd
+import numpy as np
 from scipy import interpolate
+from typing import Union, List, Tuple, TypeVar
 
+ParameterType = TypeVar('ParameterType', List[List[str]], List[Tuple[str, str, str]])
 
-class Interpolation:
-    def __init__(self, data, categorical_parameters, continuous_parameters, order, func=None):
-        data = data
-        self.key_columns = categorical_parameters
 
-        if data.empty:
-            raise ValueError("Must supply some input data")
+class Interpolation:
+    """A callable that returns the result of an interpolation function over input data.
 
-        self.parameter_columns, self_data = validate_parameters(data, continuous_parameters, order)
-        self.func = func
+        Attributes
+        ----------
+        data :
+            The data from which to build the interpolation. Contains
+            cateogrical_parameters and continuous_parameters.
+        categorical_parameters :
+            Column names to be used as categorical parameters in Interpolation
+            to select between interpolation functions.
+        continuous_parameters :
+            Column names to be used as continuous parameters in Interpolation. If
+            bin edges, should be of the form (column name used in call, column name
+            for left bin edge, column name for right bin edge). If order is 0 and
+            any continuous parameter is given without bin edges, continuous bins
+            are created using the given parameter as midpoints.
+        order :
+            Order of interpolation.
+        """
 
-        if len(self.parameter_columns) not in [1, 2]:
-            raise ValueError("Only interpolation over 1 or 2 variables is supported")
+    def __init__(self, data: pd.DataFrame, categorical_parameters: Union[List[str], Tuple[str]],
+                 continuous_parameters: ParameterType, order: int, extrapolate: bool):
+        # TODO: allow for order 1 interpolation with binned edges
+        if order != 0:
+            raise NotImplementedError(f'Interpolation is only supported for order 0. You specified order {order}')
```

Comment:
I'm a little bit confused with this line. Kate explained why she kept it here. I'm just leaving a comment in case that it's not just me who get confused by this check.

Reply:
Yeah this is probably sufficient without the additional check in validate_params for order in [0, 1]. Basically, what I was thinking was that while this is currently redundant, at some point, we'll want to remove this restriction that order has to be 0, at which point, we'll want the later check (which is what was originally in the interpolation).

Reply:
I got rid of the later redundant check + all the other dead order 1 code so it should be cleaner and less confusing now